event-handlers/src/test/java/no/sikt/nva/nvi/events/UpsertNviCandidateHandlerTest.java

```diff
@@ -91,40 +88,22 @@ void shouldSaveNewNviCandidateWithPendingInstitutionApprovalsIfCandidateDoesNotE
         var sqsEvent = createEvent(identifier, creators, instanceType, randomLevel, publicationDate, institutionPoints);
         handler.handleRequest(sqsEvent, CONTEXT);
 
-        var expectedCandidate = createExpectedCandidate(identifier, creators, instanceType, randomLevel,
-                                                        publicationDate, institutionPoints);
-        var fetchedCandidate = nviService.findByPublicationId(expectedCandidate.publicationId())
-                                   .map(Candidate::candidate);
+        var fetchedCandidate = CandidateBO.fromRequest(() -> generatePublicationId(identifier), candidateRepository,
+                                                       periodRepository).toDto();
 
-        assertThat(fetchedCandidate.orElseThrow(), is(equalTo(expectedCandidate)));
+        assertThat(fetchedCandidate.approvalStatuses().get(0).institutionId(), is(equalTo(institutionId)));
```

Comment:
Assert that the all the data in fetchedCandidate is as expected? Compare objects?